controllers/encryption.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2021 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package controllers
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/stolostron/go-template-utils/v2/pkg/templates"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	policyv1 "github.com/stolostron/config-policy-controller/api/v1"
+)
+
+const IVAnnotation = "policy.open-cluster-management.io/encryption-iv"
+
+// getEncryptionKey will get the encryption key in the managed cluster namespace used for policy template encryption.
+func (r *ConfigurationPolicyReconciler) getEncryptionKey(namespace string) (*cachedEncryptionKey, error) {
+	// #nosec G101
+	const secretName = "policy-encryption-key"
+
+	ctx := context.TODO()
+	objectKey := types.NamespacedName{
+		Name:      secretName,
+		Namespace: namespace,
+	}
+	encryptionSecret := &corev1.Secret{}
+
+	err := r.Get(ctx, objectKey, encryptionSecret)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get the encryption key from Secret %s/%s: %w", namespace, secretName, err)
+	}
+
+	key := &cachedEncryptionKey{
+		key:         encryptionSecret.Data["key"],
+		previousKey: encryptionSecret.Data["previousKey"],
+	}
+
+	return key, nil
+}
+
+// getEncryptionConfig returns the encryption config for decrypting values that were encrypted using Hub templates.
+// The returned bool determines if the cache was used. This value can be helpful to know if the cache should be
+// refreshed if the decryption failed. The forceRefresh argument will skip the cache and update it with what the
+// API returns.
+func (r *ConfigurationPolicyReconciler) getEncryptionConfig(policy policyv1.ConfigurationPolicy, forceRefresh bool) (
+	templates.EncryptionConfig, bool, error,
+) {
+	log := log.WithValues("policy", policy.GetName())
+	var usedCache bool
```

Comment:
does bool  default to false here , i see it being used before it is set.  would you consider setting it to false anyway for clarity

Reply:
or may be add a comment that it false by default